docs/benchmarks-design.md

```diff
@@ -0,0 +1,324 @@
+# <!-- omit in toc --> Superbenchmark Benchmarks Abstraction
+
+## <!-- omit in toc -->Table of Content
+- [Introduction](#introduction)
+  - [Overview](#overview)
+  - [Goals](#goals)
+- [High-level Technical Design](#high-level-technical-design)
+- [Detailed Technical Design](#detailed-technical-design)
+  - [ModelBase Benchmarks](#modelbase-benchmarks)
+    - [Training](#training)
+    - [Inference](#inference)
+  - [Micro Benchmarks](#micro-benchmarks)
+  - [DockerBase Benchmarks](#dockerbase-benchmarks)
+  - [Benchmark Registry](#benchmark-registry)
+    - [Design](#design)
+    - [Examples](#examples)
+- [Interfaces](#interfaces)
+  - [Inputs](#inputs)
+  - [Invoke](#invoke)
+  - [Outputs](#outputs)
+    - [Design](#design-1)
+    - [Example](#example)
+- [Appendix](#appendix)
+
+## Introduction
+
+### Overview
+
+Superbenchark is a benchmark framework that contains lots of useful micro-benchmarks and E2E-benchmarks targeting for hardware/software validation,  performance measurement and offering advice for better resource utilization. For now, Superbenchmark suppert:
+
+Micro-benchmark: FIO, P2P_Bandwidth, TensorCore_Performance, GPU_FLOPS, TCP_Performance, RDMA_Performance, MPI_Performance, NCCL_Performance, Stress_Test.
+
+E2E-benchmark: 32-CNN models, LSTM, BERT, Seq2Seq, DeepSpeech2, NASNet, GraphSage
+
+However the current structure of all the benchmarks are very loose. The pain points we have met are as follows:
```

Comment:
Please remove all before pain point and change to this paragraph.
`SuperBench is a benchmark framework that contains lots of useful micro-benchmarks and E2E-benchmarks targeting for hardware/software validation. However, we met lots of issues when developing SuperBench v0.1 for prototyping. Therefore, we summarize our pain points as follows:`